src/include/sof/audio/host_copier.h

```diff
@@ -0,0 +1,91 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Baofeng Tian <baofeng.tian@intel.com>
+ */
+
+/**
+ * \file audio/host_copier.h
+ * \brief host copier shared header file
+ * \authors Baofeng Tian <baofeng.tian@intel.com>
+ */
+
+#ifndef __SOF_host_COPIER_H__
+#define __SOF_host_COPIER_H__
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component_ext.h>
+#include <sof/audio/pcm_converter.h>
+#include <sof/lib/dma.h>
```

Comment:
here you need headers for macros, typedefs and structures, that are embedded in structures, defined here. Whereas you don't need headers if just a pointer is used. E.g. you don't need buffer.h, it would be enough to forward-declare `struct comp_buffer` here

Reply:
ok, I did not realize such rules before, let me have a try and update

Reply:
to clarify - those rules aren't some style rules, they're needed to compile the code. If a structure just has a pointer to another structure you don't need to know what that structure contains to compile this, all pointers are the same. But if it contains a whole structure embedded in it, then you need its definition to compile that. Same with macros and typedefs. But with typedefs it's probably the same - if you just have a pointer to such a typedef, maybe you can just forward-declare it.